letsencrypt-nginx/letsencrypt_nginx/nginxparser.py

```diff
@@ -1,56 +1,241 @@
+                val = self.peeked
+                self.peeked = None
+                return val
+            val = self.src.read(1)
+            if val == '':
+                self.eof = True
+                return None
+            return val
+
+        def parse_file(self):
+            """Tries to parse the file and returns a list of blocks parsed."""
+
+            result = self._parse_block()
+            if not self.eof:
+                raise ParseException("Invalid configuration file")
+
+            return result
+
+        def _parse_block(self):
+            result = []
+            while True:
+                res = self._parse_next()
+                if res is None:
+                    return result
+                result.append(res)
+
+        def _is_whitespace(self, val, includeNL):
+            if includeNL:
+                return val in [' ', '\t', '\n', '\r']
+            return val in [' ', '\t']
+
+        def _parse_whitespace(self, includeNL):
+            if self._is_whitespace(self._peek(), includeNL):
+                return self._read()
+            return None
+
+        def _is_at_newline(self):
+            return self._peek() in ['\n', '\r']
+
+        def _read_newline(self):
+            if self._read() == '\r' and self._peek() == '\n':
+                self._expect('\n')
+
+        def _parse_newline(self):
+            if self._is_at_newline():
+                self._read_newline()
+                return []
+            return None
+
+        def _collect_until_end_of_line(self):
+            result = ''
+
+            while not self._is_at_newline():
+                result = result + self._read()
+
+            self._read_newline()
+
+            return result
+
+        def _parse_comment(self):
+            if self._peek() == '#':
+                self._expect('#')
+                return ['#', self._collect_until_end_of_line()]
+            return None
+
+        def _is_key_char(self, val):
+            return val.isalnum() or val == '_' or val == '/'
+
+        def _parse_key(self):
+            result = ''
+            while True:
+                val = self._peek()
+                if val is None or not self._is_key_char(val):
+                    if result == '':
+                        return None
+                    return result
+                result = result + self._read()
+
+        def _parse_string(self):
+            startVal = self._read()
+            result = startVal
+            while True:
+                val = self._read()
+                result = result + val
+                if val == None:
+                    raise ParseException("Invalid configuration file, unfinished string literal")
+                if val == startVal:
+                    return result
+
+        def _parse_value(self):
+            self._pass_whitespace(True)
+            result = ''
+            while True:
+                val = self._peek()
+                if val is None or val in ['{', ';']:
```

Comment:
You could just say `val in '{;'`, since a string is a sequence of chars. (And it'd likely get interned, saving a list construction, but that's probably academic.)